examples/autopilot_server/autopilot_server.cpp

```diff
@@ -0,0 +1,312 @@
+    // thread as a ground station.
+    std::thread autopilotThread([]() {
+        mavsdk::Mavsdk mavsdkTester;
+        mavsdk::Mavsdk::Configuration configuration(
+            mavsdk::Mavsdk::Configuration::UsageType::Autopilot);
+        mavsdkTester.set_configuration(configuration);
+
+        auto result = mavsdkTester.add_any_connection("udp://127.0.0.1:14551");
+        if (result == mavsdk::ConnectionResult::Success) {
+            std::cout << "Connected autopilot server side!" << std::endl;
+        }
+
+        auto prom = std::promise<std::shared_ptr<mavsdk::System>>{};
+        auto fut = prom.get_future();
+        mavsdkTester.subscribe_on_new_system([&mavsdkTester, &prom]() {
+            std::cout << "Discovered MAVSDK GCS" << std::endl;
+            auto system = mavsdkTester.systems().back();
+            prom.set_value(system);
+            mavsdkTester.subscribe_on_new_system(nullptr);
+        });
+
+        std::cout << "Sleeping thread " << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(5));
+
+        if (mavsdkTester.systems().size() == 0) {
+            std::cout << "No System Found from Autopilot" << std::endl;
+            return;
+        } else {
+            std::cout << "Setting System" << std::endl;
+        }
+        auto system = mavsdkTester.systems().back();
+        if (system == nullptr) {
+            std::cout << "System is null??" << std::endl;
+        }
+
+        // Create server plugins
+        auto paramServer = mavsdk::ParamServer{system};
+        auto telemServer = mavsdk::TelemetryServer{system};
+        auto actionServer = mavsdk::ActionServer{system};
+
+        // These are needed for MAVSDK at the moment
+        paramServer.provide_param_int("CAL_ACC0_ID", 1);
+        paramServer.provide_param_int("CAL_GYRO0_ID", 1);
+        paramServer.provide_param_int("CAL_MAG0_ID", 1);
+        paramServer.provide_param_int("SYS_HITL", 0);
+        paramServer.provide_param_int("MIS_TAKEOFF_ALT", 0);
+        // Add a custom param
+        paramServer.provide_param_int("my_param", 1);
+
+        // Allow the vehicle to change modes, takeoff and arm
+        actionServer.set_allowable_flight_modes({true, true, true});
+        actionServer.set_allow_takeoff(true);
+        actionServer.set_armable(true, true);
+
+        // Create a mission raw server
+        // This will allow us to receive missions from a GCS
+        auto missionraw = mavsdk::MissionRawServer{system};
+        std::cout << "MissionRawServer created" << std::endl;
+
+        auto mission_prom = std::promise<MissionRawServer::MissionPlan>{};
+        missionraw.subscribe_incoming_mission(
+            [&mission_prom](MissionRawServer::Result res, MissionRawServer::MissionPlan plan) {
+                std::cout << "Received Uploaded Mission!" << std::endl;
+                std::cout << plan << std::endl;
+                mission_prom.set_value(plan);
```

Comment:
This will crash when uploading a second mission. Either unsubscribe or at least add a comment.